### PR TITLE
Fire hook on completion

### PR DESCRIPTION
**Pull request by ExternalReality**
When `juju upgrade-series complete <machine>` is called the uniter should fire completion hooks.

### CODE REVIEW

**Comment by ExternalReality** [comment]
!!build!!

**Comment by ExternalReality** [comment]
[Ensure Uniter can fire `post-series-upgrade` hooks and takethe appropriate "around"  actions.](https://trello.com/c/tEnLCgru/389-ensure-uniter-can-fire-post-series-upgrade-hooks-and-takethe-appropriate-around-actions)

**Comment by manadart** [comment]
Does this one need rebasing now that https://github.com/juju/juju/pull/8963 is landed?

File: api/common/upgradeseries.go [hunk 1]
```diff
@@ -48,13 +51,23 @@ func (u *UpgradeSeriesAPI) WatchUpgradeSeriesNotifications() (watcher.NotifyWatc
 	return w, nil
 }
 
-// UpgradeSeriesStatus returns the upgrade series status of a unit from remote state
-func (u *UpgradeSeriesAPI) UpgradeSeriesStatus() (string, error) {
+// UpgradeSeriesPrepareStatus returns the upgrade series status of a unit from remote state
```
Comment (manadart) [inline api/common/upgradeseries.go]: Wrong method name.

I know the prepare/complete status is set to converge, it seems janky to have 2 methods at he API server that hide passing the status type, and then to expose the type client side and call the server methods that hide it.

File: api/uniter/unit.go [hunk 2]
```diff
@@ -648,14 +649,14 @@ func (u *Unit) WatchUpgradeSeriesNotifications() (watcher.NotifyWatcher, error)
 	return u.st.WatchUpgradeSeriesNotifications()
 }
 
-// UpgradeSeriesPrepareStatus returns the upgrade series status of a unit from remote state
-func (u *Unit) UpgradeSeriesStatus() (string, error) {
-	return u.st.UpgradeSeriesStatus()
+// UpgradeSeriesStatus returns the upgrade series status of a unit from remote state
```
Comment (manadart) [inline api/uniter/unit.go]: And then back to the 2 method style here.

**Comment by ExternalReality** [comment]
$$MERGE$$

**Pull request merged**